src/integrationTest/java/org/opensearch/security/FlsDlsAndFieldMaskingTest.java

```diff
@@ -0,0 +1,946 @@
+			Song song = FIRST_INDEX_SONGS_BY_ID.get(songId);
+
+			SearchRequest searchRequest = searchRequestWithScroll(FIRST_INDEX_NAME, 1);
+			searchRequest.source().sort("_id", SortOrder.ASC);
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, containNotEmptyScrollingId());
+
+			SearchScrollRequest scrollRequest = getSearchScrollRequest(searchResponse);
+
+			SearchResponse scrollResponse = restHighLevelClient.scroll(scrollRequest, DEFAULT);
+			assertThat(scrollResponse, isSuccessfulSearchResponse());
+			assertThat(scrollResponse, containNotEmptyScrollingId());
+			assertThat(searchResponse, searchHitsContainDocumentWithId(0, FIRST_INDEX_NAME, songId));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_TITLE, song.getTitle()));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_LYRICS, VALUE_TO_MASKED_VALUE.apply(song.getLyrics())));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_ARTIST, VALUE_TO_MASKED_VALUE.apply(song.getArtist())));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_STARS, song.getStars()));
+		}
+
+		//DLS
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(TWINS_FIRST_ARTIST_READER)) {
+			SearchRequest searchRequest = searchRequestWithScroll(FIRST_INDEX_NAME, 2);
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, containNotEmptyScrollingId());
+
+			SearchScrollRequest scrollRequest = getSearchScrollRequest(searchResponse);
+
+			SearchResponse scrollResponse = restHighLevelClient.scroll(scrollRequest, DEFAULT);
+			assertThat(scrollResponse, isSuccessfulSearchResponse());
+			assertThat(scrollResponse, containNotEmptyScrollingId());
+			assertThat(searchResponse, numberOfTotalHitsIsEqualTo(1));
+			assertThat(searchResponse, searchHitContainsFieldWithValue(0, FIELD_ARTIST, ARTIST_TWINS));
+		}
+	}
+
+	@Test
+	public void aggregateDataAndComputeAverage() throws IOException {
+		//FIELD MASKING
+		try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(MASKED_ARTIST_LYRICS_READER)) {
+			String aggregationName = "averageStars";
+			SearchRequest searchRequest = averageAggregationRequest(FIRST_INDEX_NAME, aggregationName, FIELD_STARS);
+
+			SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+
+			assertThat(searchResponse, isSuccessfulSearchResponse());
+			assertThat(searchResponse, containAggregationWithNameAndType(aggregationName, "avg"));
+			Aggregation actualAggregation = searchResponse.getAggregations().get(aggregationName);
+			assertThat(actualAggregation, instanceOf(ParsedAvg.class));
+			assertThat(((ParsedAvg) actualAggregation).getValue(), not(Double.POSITIVE_INFINITY));
```

Comment:
what is the expected value here?

Reply:
User has access to the stars field, so the average value should be calculated. I've changed the assertion: https://github.com/opensearch-project/security/pull/2258/commits/e882b4a79e2657b6dd4c1a173fef2c4154b056b7